skip complex logaddexp tests in 3.12+

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #140733
* #139533
* __->__ #140731

This test is failing locally in 3.12 and 3.13 and is blocking 3.13 CI enablement.

It may have to do with scipy version, see .ci/docker/requirements-ci.txt (3.12+ has scipy 1.12.0/1.14.1, where as < 3.12 requires scipy 1.10.1).

Wanted to xfail these tests, but they somehow pass sometimes on CI.

cc @ezyang @anjali411 @dylanbespalko @mruberry @nikitaved @amjames

williamwen42 committed:
xfail complex logaddexp tests

[ghstack-poisoned]

ezyang reviewed (approved):
I don't really understand though why is there not a python version test here

williamwen42 committed:
Update on "xfail complex logaddexp tests in 3.12+"


This test is failing locally and is blocking 3.13 CI enablement.

cc ezyang anjali411 dylanbespalko mruberry nikitaved amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "xfail complex logaddexp tests in 3.12+"


This test is failing locally in 3.12 and 3.13 and is blocking 3.13 CI enablement.

It may have to do with scipy version, see .ci/docker/requirements-ci.txt (3.12+ has scipy 1.12.0/1.14.1, where as < 3.12 requires scipy 1.10.1).

cc ezyang anjali411 dylanbespalko mruberry nikitaved amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "xfail complex logaddexp tests in 3.12+"


This test is failing locally in 3.12 and 3.13 and is blocking 3.13 CI enablement.

It may have to do with scipy version, see .ci/docker/requirements-ci.txt (3.12+ has scipy 1.12.0/1.14.1, where as < 3.12 requires scipy 1.10.1).

cc ezyang anjali411 dylanbespalko mruberry nikitaved amjames

[ghstack-poisoned]

williamwen42 committed:
Update on "skip complex logaddexp tests in 3.12+"


This test is failing locally in 3.12 and 3.13 and is blocking 3.13 CI enablement.

It may have to do with scipy version, see .ci/docker/requirements-ci.txt (3.12+ has scipy 1.12.0/1.14.1, where as < 3.12 requires scipy 1.10.1).

Wanted to xfail these tests, but they somehow pass sometimes on CI.

cc ezyang anjali411 dylanbespalko mruberry nikitaved amjames

[ghstack-poisoned]

williamwen42 commented:
@pytorchbot merge

malfet commented on test/test_binary_ufuncs.py:
```diff
@@ -3571,6 +3572,8 @@ def _test_helper(a, b):
     )
     def test_logaddexp(self, device, dtype):
+        if sys.version_info >= (3, 12) and dtype in (torch.complex64, torch.complex128):
+            return self.skipTest("complex flaky in 3.12")
```
Do you mind creating a TODO issues and referencing it here?
